src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepExecution.java

```diff
@@ -71,27 +41,20 @@ public boolean start() throws Exception {
 
     @Override
     public void stop(@NonNull Throwable cause) throws Exception {
-        if (agent != null) {
-            agent.stop(listener);
-            listener.getLogger().println(Messages.SSHAgentBuildWrapper_Stopped());
-        }
-        purgeSockets();
-    }
-
-    @Override
-    public void onResume() {
-        super.onResume();
         try {
-            purgeSockets();
-            initRemoteAgent();
```

Comment:
This never made sense, and probably contributed to the instability of the plugin. Why would we stop and restart the SSH agent process, running on some agent computer, just because the controller restarted?